LOS_Interface/Inc/config.hpp

```diff
@@ -11,5 +16,16 @@ const uint8_t GIMBAL_GRABBER_TOGGLE_INDEX = 5; // three-way switch for gimbal, g
 const uint8_t LEFT_GIMBAL_GRABBER_CRANE = 6;
 const uint8_t RIGHT_GIMBAL_GRABBER_MOUTH = 7;
 
+/* LOS_Actuators */
+const uint8_t NUM_CHANNELS = 2;
```

Comment:
ah so NUM_CHANNELS needs to be changed when we add a new channel....

what happens when num_channels doesn't correspond with how many channels we actually have?

Reply:
If you add too many channels, it would cause a compile error (Since the number of channels in the motor_channels_ array would be too many). If you undershoot, you would end up with a seg fault if you try to access a channel outside of those defined in the array. Just as an example:

```
#define A 3

int x = 1;
int y = 2;
int z = 3;
int *a[A] = {&x, &y};

int main(int argc, char const *argv[])
{
  printf("%d, %d, %d\r\n", *a[0], *a[1], *a[2]);
  return 0;
}
`
This would segfault in a similar way to the motor channels if we overdefine NUM_CHANNELS and then access one of the non initialized pointers.

Reply:
Having a "registration" function that each motor channel calls to add itself to a vector that holds the channels would solve this issue since we could just use the current vector size as the number of valid channels and it would ofc grow and shrink automatically as we add and remove motor channels in the config.

Reply:
I'm also going to cc @Aidan-B .


It'd be nice if we could either use an auto-sizing vector OR check for mem errors on initialization (so it doesn't happen while flying).

Reply:
I'm wondering if it needs to be that intricate or if we just do like a 
```
static_assert(motor_channels_[NUM_CHANNELS - 1] != nullptr)
```
Like, do we need to a full like idiot proof check on it or can we assume that no one is gonna go in and set a value in the motor config to nullptr. I guess there's no reason not to do that at the end of the day tho